core/src/packet_forward_manager.rs

```diff
@@ -0,0 +1,427 @@
+use {
+    crate::{
+        banking_stage::TOTAL_BUFFERED_PACKETS, qos_service::QosService,
+        unprocessed_packet_batches::*,
+    },
+    solana_perf::packet::Packet,
+    solana_runtime::{bank::Bank, cost_model::TransactionCost, cost_tracker::CostTracker},
+    solana_sdk::transaction::SanitizedTransaction,
+    std::sync::Arc,
+};
+
+#[derive(Debug)]
+pub struct PacketForwardManager<'a> {
+    cost_trackers: Vec<CostTracker>,
+    forwardable_packets: Vec<Vec<&'a Packet>>,
```

Comment:
This seems like it could be condensed into a single `Vec<ForwardableBlock>`, where a `ForwardableBlock` looks like:

```
struct ForwardableBlock {
  cost_tracker: CostTracker,
  forwardable_packets: Vec<&'a Packet>
```